index/scorch/merge.go

```diff
@@ -263,6 +263,17 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot,
 	}
 
 	for _, notification := range notifications {
+		// update the idex snapshot reference counts for all
+		// the merge introductions which are already done.
+		if len(notification) == cap(notification) {
+			newSnapshot := <-notification
+			atomic.AddUint64(&s.stats.TotFileMergeIntroductionsDone, 1)
+			if newSnapshot != nil {
+				_ = newSnapshot.DecRef()
+			}
+			continue
+		}
+
 		select {
 		case <-s.closeCh:
 			atomic.AddUint64(&s.stats.TotFileMergeIntroductionsSkipped, 1)
```

Comment:
One thought... instead, if we end up here in this closeCh case... perhaps this would be an additional place to invoke the newSnapshot.DecRef()?

Reply:
indeed, I had that thought.. But couldn't visualise it fully - as we may again meet the scenario where we wait for reading from the notification channel?  newSnapshot := <-notification and a bail out needed there too..